src/components/implementation/mem_mgr/naive/mem_man.c

```diff
@@ -408,7 +408,7 @@ int __mman_fork_spd(spdid_t spd, u32_t s_spd_d_spd, vaddr_t base, u32_t len)
 	 * are needed in a dependent of d_spd? */
 
 	for ( s_addr = base ; !ret && s_addr < s_addr + len; s_addr += PAGE_SIZE ) {
-		if (!(m = mapping_lookup(s_spd, s_addr))) continue;
+		if (!(m = mapping_lookup(s_spd, s_addr))) continue;							// why not bug?
```

Comment:
Perfect question for your new-found research log ;-)

What if the user passed in an invalid range of addresses?  Should we `BUG` here because the user-passed something erroneous in?

If you mean "why not trigger failure" instead, then the question is one of what the point of this function is.  This code copies a range of VAS.  If there are holes within the range, it skips over them.  That is neither wrong, or right.  It is simply what the function does.  You could change it to have different behavior, but you'd have to understand the trade-off your making.